src/nvim/strings.c

```diff
@@ -45,6 +45,41 @@
 #include "nvim/os/os.h"
 #include "nvim/os/shell.h"
 
+/// Compares two strings for equality.
+bool strequal(const char *a, const char *b)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strcmp(a, b) == 0;
+}
+
+/// Compares two sized strings for equality.
+bool strnequal(const char *a, const char *b, size_t size)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strncmp(a, b, size) == 0;
+}
+
+/// Compares two strings for equality ignoring differences in case.
+bool striequal(const char *a, const char *b)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strcasecmp(a, b) == 0;
+}
+
+/// Compares two sized strings for equality ignoring differences in case.
+bool strniequal(const char *a, const char *b, size_t size)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strncasecmp(a, b, size) == 0;
+}
+
+/// Checks a string for being NULL or empty.
+bool strempty(const char *s)
+  FUNC_ATTR_PURE
+{
+  return s == NULL || s[0] == '\0';
+}
```

Comment:
And how about making this and `strn?equal` functions `static inline` in the header? This way compiler will be able to optimize them out (not suggesting for `strn?iequal` because I do not see gcc optimizing even `strcasecmp` out: most likely due to it being locale-dependent I think).